### PR TITLE
[expo-development-client] Polish launcher UI

### PR DESCRIPTION
**Pull request by nikki93**
# Why

The development client launcher UI is currently pretty barebones. This is also because some of the functionality such as using the system QR code scanner or listing release channels hasn't been built out yet. We do have a design for the current functionality at: https://www.figma.com/file/Mcwt6YeqUkQe82lc4mzNNk/Expo-Go-%26-Dev-client?node-id=1%3A18 -- this change uses that design.

# How

Just writing some React Native code to mirror the above design.

# Test Plan

Try with `useDevClient = true` (iOS, AppDelegate.swift) or `USE_DEV_CLIENT = true;` (Android, MainApplication.java) in 'bare-expo'.

### CODE REVIEW

File: packages/expo-development-client/bundle/App.tsx
Comment (lukmccall) [inline packages/expo-development-client/bundle/App.tsx:21]: Won't be easier to pass the `setLoading` function to `loadAppFromUrl`?

**Review by lukmccall — APPROVED** [review]
LGTM 🚀

File: packages/expo-development-client/bundle/App.tsx
Comment (nikki93) [reply packages/expo-development-client/bundle/App.tsx:21]: Good point, I'll change it to be that way.

**Comment by nikki93** [comment]
@lukmccall I'll wait until we land your Android bridge PR before merging this, so that I can generate the JS bundle again and test on new Android bridge too (most likely it just works :) )

**Pull request merged**
